### ISSUE TITLE
Conflict between IDE0079 and other rules

### ISSUE THREAD

**Issue by lonix1** [dotnet/roslyn#70201]
### Analyzer

**Diagnostic ID**: [IDE0079](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0079): `Remove unnecessary suppression`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://learn.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 7.0.400](https://dotnet.microsoft.com/download/dotnet/7.0)

### Describe the bug

```cs
#pragma warning disable CA1812

namespace Foo;
internal sealed class Foo {
  //...
}
```

That internal class is instantiated in a way that the rule doesn't detect.

But I get `CA1812 : Avoid uninstantiated internal classes`. So I add `#pragma warning disable CA1812`.

But then I get `IDE0079 : Remove unnecessary suppression`.

So I remove it. And then I get `CA1812`.

So I re-add it, remove it, re-add it, remove it, ...! :smile:

### Steps To Reproduce

See above

### Expected behavior

No warning

### Actual behavior

Warning

**Comment by mavasani** [comment]
I am unable to repro the original issue reported by @lonix1. `CA1812` is a compilation end analyzer and we explicitly skip all compilation end diagnostics from IDE0079 analysis: https://github.com/dotnet/roslyn/blob/005c4c114880a26e57fd4b19aa4167a1b25bc6da/src/Analyzers/Core/Analyzers/RemoveUnnecessarySuppressions/AbstractRemoveUnnecessaryPragmaSuppressionsDiagnosticAnalyzer.cs#L431

**Comment by mavasani** [comment]
> I'm also getting this behaviour for `CA1308` "replace the call to 'ToLowerInvariant' with 'ToUpperInvariant'". If I suppress that rule, I get a warning to remove the suppression, and when I do, it warns me to fix it, etc.
> 
> I suspect there are other conflicts as well. I don't remember in which version update this issue started.
> 
> # UPDATE
> I'm getting this for _many_ different types of rules.

I am unable to repro this issue either

![IDE0079_NoRepro](https://github.com/dotnet/roslyn/assets/10605811/ce1651e3-6535-4412-b2ed-3c5afa7d603b)

**Comment by mavasani** [comment]
@lonix1 Can you please let us know which version of VS are you seeing the issue?
